lib/rust/ensogl/core/src/display/world.rs

```diff
@@ -534,10 +526,12 @@ impl WorldData {
     }
 
     fn run_stats(&self, time: Duration) {
-        let previous_frame_stats = self.stats.begin_frame(time);
-        if let Some(stats) = previous_frame_stats {
-            self.on.prev_frame_stats.run_all(&stats);
+        self.stats.calculate_prev_frame_fps(time);
+        {
+            let stats_borrowed = self.stats.borrow();
+            self.on.prev_frame_stats.run_all(&stats_borrowed.stats_data);
         }
+        self.stats.reset_per_frame_statistics();
     }
```

Comment:
I know it's from older code, but I don't quite get what "running stats" means in this context. It is about displaying them?

Reply:
where is "running stats" there? `run_all` refers to running all registered callbacks.